portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro

```diff
@@ -191,4 +404,104 @@ definition {
 		AssertLocation.assertPartialLocation(value1 = "${specificURL}");
 	}
 
+	@description = "Execute login from SP."
+	macro SSOLoginToSP {
+		Navigator.openSpecificURL(url = "${url}");
+
+		if (IsElementPresent(locator1 = "UserBar#USER_AVATAR_IMAGE")) {
+			User.logoutPG();
+		}
+
+		User.loginPG(
+			password = "test",
+			specificURL = "${url}",
+			userEmailAddress = "${emailAddress}");
+	}
+
+	@description = "Update IDP Connection data at SP."
+	macro updateIDPConnectionConfiguration {
+		Navigator.openSpecificURL(url = "${url}");
+
+		User.loginPG(
+			password = "test",
+			specificURL = "${url}",
+			userEmailAddress = "${emailAddress}");
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Security",
+			panel = "Control Panel",
+			portlet = "SAML Admin");
+
+		SAMLNavigation.gotoIdentityProviderConnections();
+
+		Click.waitForMenuToggleJSClick(
+			key_samlEntityID = "${idpConnectionName}",
+			locator1 = "CPSAMLAdmin#SAML_CONNECTION_ACTION_BUTTON");
+
+		MenuItem.click(menuItem = "Edit");
+
+		if (isSet(nameIdentifierFormat)) {
+			Select(
+				locator1 = "Select#SAML_NAME_IDENTIFIER_FORMAT_SELECT",
+				value1 = "${nameIdentifierFormat}");
+		}
+
+		if (isSet(idpAttributesMapping)) {
+			Check(locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_ENABLED_CHECKBOX");
```

Comment:
What does setting idpAttrbutesMapping have to do with checking the enabled checkbox?